runtime/lua/vim/shared.lua

```diff
@@ -801,7 +801,34 @@ end
 ---@return string String with whitespace removed from its beginning and end
 function vim.trim(s)
   vim.validate('s', s, 'string')
-  return s:match('^%s*(.*%S)') or ''
+  local len = #s
+  local byte = string.byte
+
+  if len == 0 then
+    return ''
+  end
+
+  -- `s` is already trimmed
+  if byte(s, 1) > 32 and byte(s, len) > 32 then
+    return s
+  end
+
+  local start = 1
+  while start <= len and byte(s, start) <= 32 do
```

Comment:
Is less `<=32` safe to do here? I'm assuming that covers `\n\t` too?

Reply:
Yes, everything below 32 is special (non printable) characters and 32 is a space.